privacyidea/lib/tokens/yubikeytoken.py

```diff
@@ -379,29 +382,23 @@ def check_yubikey_pass(passw):
         res = False
 
         token_list = []
+        token_candidate_list = []
 
         # strip the yubico OTP and the PIN
-        modhex_serial = passw[:-32][-16:]
-        try:
-            serialnum = "UBAM" + modhex_decode(modhex_serial)
-        except TypeError as exx:  # pragma: no cover
-            log.error("Failed to convert serialnumber: %r" % exx)
-            return res, opt
-
-        # build list of possible yubikey tokens
-        serials = [serialnum]
-        for i in range(1, 3):
-            serials.append("%s_%s" % (serialnum, i))
+        prefix = passw[:-32][-16:]
 
         from privacyidea.lib.token import get_tokens
         from privacyidea.lib.token import check_token_list
-        for serial in serials:
-            tokenobject_list = get_tokens(serial=serial)
-            token_list.extend(tokenobject_list)
+
+        token_candidate_list = get_tokens(tokentype='yubikey')
```

Comment:
I am not content with this. This will fetch all yubikey tokens from the database. This could be thousand tokens. But the get_tokens() function is limited. It can not match for a tokeninfo.
But it should be possible to enhance the get_tokens() to match for certain tokeninfos. (but this will be another commit)